CONTRIBUTING.md

```diff
@@ -462,6 +462,35 @@ Note that the test environment uses a temporary directory for defining the
 SQLite databases which will be cleared each time before the group of test
 commands are invoked.
 
+#### Typing
+
+To ensure clarity, consistency, all readability, _all_ new functions should use
```

Comment:
Personally I'm happy with this. Can we kick off a `[DISCUSS]` thread on the mailing list though, making sure we have consensus?